packages/dynamoose/lib/Model/index.ts

```diff
@@ -199,23 +199,25 @@ export class Model<T extends ItemCarrier = AnyItem> extends InternalPropertiesCl
 					return result;
 				}, {});
 			},
-			"convertObjectToKey": (key: InputKey): KeyObject => {
+			"convertObjectToKey": async (key: InputKey): Promise<KeyObject> => {
 				let keyObject: KeyObject;
 				const hashKey = this.getInternalProperties(internalProperties).getHashKey();
 				if (typeof key === "object") {
+					// If we passed aliased attribute names, we need to get them back to the key names
+					const mappedKey = await this.Item.objectFromSchema(key, this, {"type": "toDynamo", "mapAttributes": true});
 					const rangeKey = this.getInternalProperties(internalProperties).getRangeKey();
 					keyObject = {
-						[hashKey]: key[hashKey]
+						[hashKey]: mappedKey[hashKey]
 					};
-					if (rangeKey && typeof key[rangeKey] !== "undefined" && key[rangeKey] !== null) {
-						keyObject[rangeKey] = key[rangeKey];
+					if (rangeKey && typeof mappedKey[rangeKey] !== "undefined" && mappedKey[rangeKey] !== null) {
+						keyObject[rangeKey] = mappedKey[rangeKey];
 					}
 				} else {
 					keyObject = {
 						[hashKey]: key
 					};
 				}
-				return keyObject;
+				return Promise.resolve(keyObject);
```

Comment:
Shouldn't need to modify this line. Returning from an `async` function is enough to have it return a promise. So `Promise.resolve` is not necessary since it's an async function.